tests/test_peak_core_and_tile.py

```diff
@@ -0,0 +1,197 @@
+        config_data, a, b, = value
+        for addr, data in config_data:
+            tester.configure(addr, data)
+        tester.poke(tester._circuit.data0, a)
+        tester.poke(tester._circuit.data1, b)
+
+    def core_output_monitor(tester, value):
+        tester.expect(tester._circuit.alu_res, value)
+
+    core = PeakCore(PE_fc)
+    core.name = lambda: "PECore"
+    circuit = core.circuit()
+
+    inputs, outputs = sequences
+    tester = BasicSequenceTester(circuit, [
+        (core_input_driver, InputSequence(inputs)),
+        (core_output_monitor, OutputSequence(outputs))
+    ], circuit.clk, circuit.reset)
+    tester.reset()
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        for filename in dw_files:
+            shutil.copy(filename, tempdir)
+
+        tester.compile_and_run("verilator",
+                               directory=tempdir,
+                               magma_opts={"coreir_libs": {"float_DW"}},
+                               flags=["-Wno-fatal"])
+
+
+def test_peak_tile_sequence(sequences, dw_files):
+    """
+    Tile level test:
+    * Generates CGRA to extract PE_tile
+    * sets tile_id input based on test app
+    * configures PE_tile based on fixed route (taken from a test app)
+    * similar input driver and output monitor behavior to core test except:
+      * inputs are driven onto the appropriate tile ports based on the fixed
+        route configuration
+      * output is similarly monitored based on the route config
+    """
+    # Use stub CGRA to get PE_tile
+    # TODO: Is there an API to just get a PE_tile?
+    io_sides = IOSide.North | IOSide.East | IOSide.South | IOSide.West
+    interconnect = create_cgra(2, 2, io_sides, num_tracks=3)
+
+    # Use test application to get a configuration
+    # TODO: We should randomize or cover all possible configurations of the
+    # CB/SBs
+    # placement, routing generated from:
+    # netlist = {
+    #     "e0": [("I0", "io2f_16"), ("p0", "data0")],
+    #     "e1": [("I1", "io2f_16"), ("p0", "data1")],
+    #     "e3": [("p0", "alu_res"), ("I2", "f2io_16")],
+    # }
+    # bus = {"e0": 16, "e1": 16, "e3": 16}
+
+    # from archipelago import pnr
+    # placement, routing = pnr(interconnect, (netlist, bus))
+    # route_config = interconnect.get_route_bitstream(routing)
+    # x, y = placement["p0"]
+
+    # placement = {'I0': (1, 0), 'I1': (2, 0), 'I2': (0, 1), 'p0': (1, 1)}
+    # routing = {'e0': [[CB_io2f_16, SB_T0_NORTH_SB_IN_B16, CB_data0]],
+    #            'e1': [[CB_io2f_16, SB_T0_NORTH_SB_IN_B16,
+    #                    SB_T0_WEST_SB_OUT_B16, RMUX_T0_WEST_B16,
+    #                    SB_T0_EAST_SB_IN_B16, CB_data1]],
+    #            'e3': [[CB_alu_res, SB_T0_WEST_SB_OUT_B16, RMUX_T0_WEST_B16, CB_f2io_16]]}
+
+    route_config = [(262401, 0), (459265, 0), (459265, 0), (327937, 2),
+                    (459009, 786432), (459009, 0)]
+    route_config = compress_config_data(route_config)
+    x, y = (1, 1)
```

Comment:
Ideally this should be read out from PnR result instead of being hardcoded.

Reply:
See above w.r.t. to the hard code port name and the routing configuration changing across test invocations